src/task/SubscribeAndReceiveSubscribeAckTask.ts

```diff
@@ -73,7 +76,24 @@ export default class SubscribeAndReceiveSubscribeAckTask extends BaseTask {
     const isSendingStreams: boolean =
       this.context.videoDuplexMode === SdkStreamServiceType.TX ||
       this.context.videoDuplexMode === SdkStreamServiceType.DUPLEX;
-    this.context.previousSdpOffer = new DefaultSDP(localSdp);
+
+    let compressedSDPOffer: Uint8Array | null;
+    const localSdpOffer = localSdp;
+
+    if (this.context.serverSupportsCompression) {
+      // If the server supports compression, then send the compressed version of the sdp
+      // and exclude the original sdp offer.
+      const prevOffer = this.context.previousSdpOffer
+        ? (<DefaultSDP>this.context.previousSdpOffer).sdp
```

Comment:
I think we generally use as instead of <>. 
```
this.context.previousSdpOffer ? (this.context.previousSdpOffer as DefaultSDP).sdp
```

Reply:
ok...will change it